src/main/java/com/spotify/reaper/storage/CassandraStorage.java

```diff
@@ -631,12 +654,17 @@ public boolean updateRepairSchedule(RepairSchedule newRepairSchedule) {
     Optional<RepairSchedule> repairSchedule = getRepairSchedule(id);
     if(repairSchedule.isPresent()){
       RepairUnit repairUnit = getRepairUnit(repairSchedule.get().getRepairUnitId()).get();
-      BatchStatement batch = new BatchStatement();
-      batch.add(deleteRepairSchedulePrepStmt.bind(repairSchedule.get().getId()));
-      batch.add(deleteRepairScheduleByClusterAndKsPrepStmt.bind(repairUnit.getClusterName(), repairUnit.getKeyspaceName(), repairSchedule.get().getId()));
-      batch.add(deleteRepairScheduleByClusterAndKsPrepStmt.bind(repairUnit.getClusterName(), " ", repairSchedule.get().getId()));
-      batch.add(deleteRepairScheduleByClusterAndKsPrepStmt.bind(" ", repairUnit.getKeyspaceName(), repairSchedule.get().getId()));
-      session.execute(batch);
+
+      session.executeAsync(deleteRepairScheduleByClusterAndKsPrepStmt
```

Comment:
batches should be a consequence and needed decision.

IMHO it'll be better to put in a _always retry_ policy into place (presuming idempotence in the driver isn't working already).